src/ceph_mon.cc

```diff
@@ -571,17 +571,16 @@ int main(int argc, const char **argv)
     string val;
     int r = store->read_meta("min_mon_release", &val);
     if (r >= 0 && val.size()) {
-      int min = atoi(val.c_str());
-      if (min &&
-	  min + 2 < (int)ceph_release()) {
+      ceph_release_t min = ceph_release_from_name(val);
+      if (min != ceph_release_t::unknown &&
+	  !can_upgrade_from(min, ceph_release())) {
```

Comment:
I'd be inclined to skip the helper function here since it's pretty trivial, introduces the possibility of getting hte arg order mixed up, and the +2 rule is explicitly encoded in the error messages anyway.

Reply:
i want to avoid treating the version as arithmetic variable. and to minimize the duplication, i revised the helper to improve the readability and to make it more useful.